src/customdocument.cpp

```diff
@@ -293,3 +302,75 @@ QMap<QString, QString> CustomDocument::parseMarkdownUrlsFromText(const QString &
 
     return urlMap;
 }
+
+void CustomDocument::moveLineUp()
+{
+    qDebug() << "Moving line up";
+    QTextCursor cursor = textCursor();
+
+    if (cursor.blockNumber() > 0) {
+        QString currentLine = cursor.block().text();
+        const int currentHorizontalPosition = cursor.positionInBlock();
+        qDebug() << "Current horizontal position: " << currentHorizontalPosition;
+
+        cursor.movePosition(QTextCursor::StartOfBlock);
+        cursor.movePosition(QTextCursor::EndOfBlock, QTextCursor::KeepAnchor);
+        cursor.removeSelectedText();
+        // also remove empty line
+        cursor.deletePreviousChar();
+
+        // cursor is now at end of prev line, move to start
+        if (!cursor.movePosition(QTextCursor::StartOfLine)) {
```

Comment:
Chaing these lines to the following solved the issue for me. Can you test?

```cpp
// cursor is now at end of prev line, move to start of block
        if (!cursor.movePosition(QTextCursor::StartOfBlock)) {
            qDebug() << "Could not move to start of block";
        }
```

Reply:
Yes! That fixed it. Thanks!

Reply:
Great. Did enough QTextCursor manipulations with the kanban feature to see the issue haha.